cassovary-core/src/main/scala/com/twitter/cassovary/graph/UndirectedGraph.scala

```diff
@@ -0,0 +1,29 @@
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+
+/*
+ * A graph where the outbound neighbors of each node equal the inbound neighbors.
+ * Given a DirectedGraph which stores the neighbors of each node (as outbound edges),
+ * this class wraps that graph in an undirected view.
+ */
+
+class UndirectedGraph(outboundGraph: DirectedGraph) extends DirectedGraph {
```

Comment:
I'd make it a trait and not override methods that do not do anything.

trait UndirectedGraph extends DirectedGraph {
def storedGraphDir = Mutual
def getNodeById(id: Int): Option[UndirectedNode]
def iterator: Iterator[UndirectedNode]
}